keps/582-preempt-based-on-flavor-order/README.md

```diff
@@ -0,0 +1,330 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a Kueue administrator I want to ensure more important jobs running on more 
+stable resources. This can happen in case that there are normal and spot instances
+in my cluster. In this case I prefer my high priority jobs not running on spot 
+instances. If high priority jobs can preempt jobs in normal ecs before try spot instances,
+stability can be achieved.
+
+My use case can be supported by set `flavorFungibility` to `BeforeNextFlavor`  in the Kueue configuration.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Cluster Queue API
+
+We extend the Cluster Queue API to introduce the new fields: flavorFungibility to opt-in and configure the new behavior.
+
+```
+type FlavorFungibility string
+
+const (
+	BeforeNextFlavor FlavorFungibility = "BeforeNextFlavor"
+	AfterAllFlavors  FlavorFungibility = "AfterAllFlavors"
+)
+
+// ClusterQueuePreemption contains policies to preempt Workloads from this
+// ClusterQueue or the ClusterQueue's cohort.
+type ClusterQueuePreemption struct {
+	...
+
+	// +kubebuilder:default="AfterAllFlavors"
+	// +kubebuilder:validation:Enum=BeforeNextFlavor;AfterAllFlavors
+	FlavorFungibility FlavorFungibility `json:"flavorFungibility"`
+}
+```
+
+### Schduling Process in Kueue
```

Comment:
You will probably depend on #806.

Reply:
yes, I think so

Reply:
Great😄. I will complete this pr in this week.